src/Init/Lean/Data/Json/Parser.lean

```diff
@@ -0,0 +1,314 @@
+/-
+Copyright (c) 2019 Gabriel Ebner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+
+Authors: Gabriel Ebner, Marc Huisinga
+-/
+prelude
+import Init.Lean.Data.Json.Basic
+import Init.Control.Except
+
+namespace Lean
+
+inductive Quickparse.Result (α : Type)
+| success {} (pos : String.Iterator) (res : α) : Quickparse.Result
+| error {} (pos : String.Iterator) (err : String) : Quickparse.Result
```

Comment:
`String.Iterator` was essential for performance in Lean 3 code, but it is mostly overhead in Lean 4. 
I have observed performance improvements in the Lean `Parser` library when I switched to `String.Pos`.
That is, `Quickparser.Result` would use `String.Pos` instead of `String.Iterator`, and `QuickParse` would be 
```lean
def Quickparse (α : Type) : Type := String → String.Pos → Quickparse.Result α
```